pkg/logutils/log.go

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package logutils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+// Pre-define log instance with default info level.
+var (
+	Logger       *zap.Logger
+	LoggerStderr *zap.Logger
+	LoggerFile   *zap.Logger
+)
+
+// LogFormat is a type help you choose the log output format, which supports "json" and "console".
+type LogFormat string
+
+const (
+	JsonLogFormat    LogFormat = "json"
+	ConsoleLogFormat LogFormat = "console"
+)
+
+// FileOutputOption supports the configuration for log file
+type FileOutputOption struct {
+	Filename   string
+	MaxSize    int
+	MaxAge     int
+	MaxBackups int
+}
+
+// LogMode is a type help you choose the log output mode, which supports "stderr","stdout","file".
+type LogMode uint32
+
+const (
+	_             LogMode = 1 << (4 - 1 - iota)
```

Comment:
(4 - 1 - iota) , is dangerous code 
what if add a new line in the future, it is easy for others to code a bug